### PR TITLE
Partial delete

### PR DESCRIPTION
**Pull request by antoine-de**
Handle IRE partial deletion

linked to https://github.com/CanalTP/navitia/pull/2061

what do you think @kinnou02 @TeXitoi @pbougue ?

### CODE REVIEW

File: kirin/ire/model_maker.py [hunk 1]
```diff
@@ -226,12 +226,29 @@ def _make_trip_update(self, vj, xml_modification):
...
+                    if not as_bool(get_value(deleted_points, 'IndicateurPRGare')):
+                        continue
+                    nav_st = self._get_navitia_stop_time(deleted_points, vj.navitia_vj)
+
+                    if nav_st is None:
+                        continue
+
+                    nav_stop = nav_st.get('stop_point', {})
+
+                    message = get_value(deleted_points, 'MotifExterne')
```
Comment (xlqian) [inline kirin/ire/model_maker.py]: `MotifExterne` is now optional, we should synchronize the merge between our PRs

File: kirin/ire/model_maker.py [hunk 2]
```diff
@@ -226,12 +226,29 @@ def _make_trip_update(self, vj, xml_modification):
...
+                    st_update = model.StopTimeUpdate(nav_stop, dep_status='delete', arr_status='delete',
```
Comment (kinnou02) [inline kirin/ire/model_maker.py]: Departure can be deleted but not the arrival, or the other way around.
`PRDebut` and `PRFin` are probably included

**Comment by antoine-de** [comment]
changes done.

linked to https://github.com/CanalTP/chaos-proto/pull/17

File: tests/integration/ire_test.py [hunk 3]
```diff
@@ -145,7 +145,7 @@ def check_db_ire_96231_normal():
         assert db_trip_delayed.contributor == 'realtime.ire'
 
 
-def check_db_ire_96231_delayed(modif_externe_is_null=False):
+def check_db_ire_96231_delayed(motif_externe_is_null=False):
```
Comment (pbougue) [inline tests/integration/ire_test.py]: :+1:

File: tests/integration/ire_test.py [hunk 4]
```diff
@@ -544,5 +544,75 @@ def test_ire_trip_without_any_motifexterne(mock_rabbitmq):
...
-    check_db_ire_96231_delayed(modif_externe_is_null=True)
+    check_db_ire_96231_delayed(motif_externe_is_null=True)
+    assert mock_rabbitmq.call_count == 1
+
+
+def test_ire_partial_removal(mock_rabbitmq):
+    """
+    the trip 840427 has been partialy deleted
+
+    Normally there is 7 stops in this VJ, but 2 (Bar-sur-Aube and Vendeuvre) have been removed
```
Comment (pbougue) [inline tests/integration/ire_test.py]: minor: there `are` 7 stops

File: tests/integration/populate_pb_test.py [hunk 5]
```diff
@@ -114,22 +116,144 @@ def test_populate_pb_with_two_stop_time():
...
+    nav vj
+    sa:1 * ---- * sa:2 * ---- * sa:3 * ---- * sa:4
+
+    we stop the vj from going to sa:3 and sa:4
+
+    sa:1 * ---- * sa:2
+
+    And we delay sa:1 of 5 minutes
+
+    Note: sa:2 departure must be removed as well as sa:3 departures/arrival and sa:4' arrival
```
Comment (pbougue) [inline tests/integration/populate_pb_test.py]: minor: maybe add that it's done in data, not by connector

**Pull request merged**